### PR TITLE
Fixed Generator::mkdir when already exists

### LINKED ISSUE #523: Warning: mkdir(): File exists
Updated to the latest version this morning and when I try and generate a new entity I get the following error:

```
[Symfony\Component\Debug\Exception\ContextErrorException]  
  Warning: mkdir(): File exists                              
                                                             
Exception trace:
 () at /var/www/orvelo2/brains-v1/vendor/sensio/generator-bundle/Generator/Generator.php:75
 Symfony\Component\Debug\ErrorHandler->handleError() at n/a:n/a
 mkdir() at /var/www/orvelo2/brains-v1/vendor/sensio/generator-bundle/Generator/Generator.php:75
 Sensio\Bundle\GeneratorBundle\Generator\Generator::mkdir() at /var/www/orvelo2/brains-v1/vendor/sensio/generator-bundle/Generator/DoctrineEntityGenerator.php:99
 Sensio\Bundle\GeneratorBundle\Generator\DoctrineEntityGenerator->generate() at /var/www/orvelo2/brains-v1/vendor/sensio/generator-bundle/Command/GenerateDoctrineEntityCommand.php:88
 Sensio\Bundle\GeneratorBundle\Command\GenerateDoctrineEntityCommand->execute() at /var/www/orvelo2/brains-v1/vendor/symfony/symfony/src/Symfony/Component/Console/Command/Command.php:256
 Symfony\Component\Console\Command\Command->run() at /var/www/orvelo2/brains-v1/vendor/symfony/symfony/src/Symfony/Component/Console/Application.php:837
 Symfony\Component\Console\Application->doRunCommand() at /var/www/orvelo2/brains-v1/vendor/symfony/symfony/src/Symfony/Component/Console/Application.php:187
 Symfony\Component\Console\Application->doRun() at /var/www/orvelo2/brains-v1/vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Console/Application.php:80
 Symfony\Bundle\FrameworkBundle\Console\Application->doRun() at /var/www/orvelo2/brains-v1/vendor/symfony/symfony/src/Symfony/Component/Console/Application.php:118
 Symfony\Component\Console\Application->run() at /var/www/orvelo2/brains-v1/application/bin/console:29
```

Knocking the require-dev version back to 3.0.10 has fixed the issue and allowed me to generate new entities.  Has there been a regression?

### PR DESCRIPTION
**Pull request by HeahDude**
Closes #523.

### CODE REVIEW

**Comment by javiereguiluz** [comment]
👍 

Status: reviewed.

@HeahDude thanks for fixing this ... and I'm sorry for having introduced it :(

File: Generator/Generator.php [hunk 1]
```diff
@@ -72,8 +72,9 @@ protected function renderFile($template, $target, $parameters)
      */
     public static function mkdir($dir, $mode = 0777, $recursive = true)
     {
-        mkdir($dir, $mode, $recursive);
-        self::writeln(sprintf('  <fg=green>created</> %s', self::relativizePath($dir)));
+        if (@mkdir($dir, $mode, $recursive)) {
```
Comment (xabbuh) [inline Generator/Generator.php]: What about checking for existence instead of silencing the operation?

File: Generator/Generator.php [hunk 2]
```diff
@@ -72,7 +72,8 @@ protected function renderFile($template, $target, $parameters)
      */
     public static function mkdir($dir, $mode = 0777, $recursive = true)
     {
-        if (@mkdir($dir, $mode, $recursive)) {
+        if (!is_dir(dirname($dir))) {
+            mkdir($dir, $mode, $recursive);
```
Comment (javiereguiluz) [inline Generator/Generator.php]: Please keep the silencing operator (Symfony always does `@mkdir`) because eventually an error will happen here.

Comment (HeahDude) [inline Generator/Generator.php]: tests are failing anyway, let's revert this one.

File: Generator/Generator.php (hunk 1 above)
Comment (fabpot) [inline Generator/Generator.php]: 👍 for @xabbuh proposal

**Comment by HeahDude** [comment]
@fabpot tests are failing when I address @xabbuh's comment

File: Generator/Generator.php [hunk 3]
```diff
@@ -72,8 +70,10 @@ protected function renderFile($template, $target, $parameters)
      */
     public static function mkdir($dir, $mode = 0777, $recursive = true)
     {
-        mkdir($dir, $mode, $recursive);
-        self::writeln(sprintf('  <fg=green>created</> %s', self::relativizePath($dir)));
+        if (!is_dir(dirname($dir))) {
```
Comment (fabpot) [inline Generator/Generator.php]: should be `$dir`, not `dirname($dir)`, no? same on the next line as well

**Comment by HeahDude** [comment]
Thanks @fabpot we're good to go now :)

**Comment by fabpot** [comment]
Thank you @HeahDude.

**Pull request merged**